### PR TITLE
Add pre-commit hook for contribute documentation.

### PR DESCRIPTION
**Pull request by reyoung**
* Also add a soft link for contributing.md to make github recognize
  this file.

### CODE REVIEW

File: doc/howto/contribute_to_paddle_en.md [hunk 1]
```diff
@@ -47,6 +47,23 @@ Then you can start to develop by making a local developement branch
 git checkout -b MY_COOL_STUFF_BRANCH
 ```
 
+## Using `pre-commit` hook
+
+Paddle developers use [pre-commit](http://pre-commit.com/) tool to manage git
+pre-commit hooks. It can help us format source codes (cpp, python), check some
+basic thing before commit (only one EOL for each file, do not add a huge file
+in git). `pre-commit` tests is a part of unit tests in Travis-CI now, every
+PR not fits hook can not be merged into Paddle.
```
Comment (gangliao) [inline doc/howto/contribute_to_paddle_en.md]: Every PR doesn't fit hook can not be merged into Paddle.

**Comment by luotao1** [comment]
#889 移动了contribute_to_paddle_en.md的路径，@gangliao 建议先merge这个PR。

File: doc/howto/contribute_to_paddle_en.md [hunk 2]
```diff
@@ -47,6 +47,23 @@ Then you can start to develop by making a local developement branch
...
+
+To use [pre-commit](http://pre-commit.com/), you should install it by
+`pip install pre-commit`, and currently, Paddle uses `clang-format` to format
+c/cpp sources. Please make sure clang-format 3.8+ installed.
+
+Then just run `pre-commit install` in your Paddle clone directory. When you
+commit your codes, the pre-commit hook will help you fit Paddle code style;
```
Comment (gangliao) [inline doc/howto/contribute_to_paddle_en.md]: commit your code, the pre-commit hook will check the local code if there is anything not suitable to commit, and so on.

**Comment by reyoung** [comment]
@gangliao done.

**Pull request merged**